cmd/argocd/commands/app.go

```diff
@@ -812,6 +826,9 @@ func setParameterOverrides(app *argoappv1.Application, parameters []string) {
 			Name:      parts[1],
 			Value:     parts[2],
 		}
+		if !argo.CheckValidParam(appSet, newParam) {
+			log.Fatalf("Parameter '%s' in '%s' does not exist in ksonnet", newParam.Name, newParam.Component)
+		}
```

Comment:
I don't see the need to do this client side validation if we are already doing this server side